src/extension/params-extractor/src/main/java/org/geoserver/params/extractor/RulesDao.java

```diff
@@ -43,40 +43,53 @@ public static String getTempRulesPath() {
 
     public static List<Rule> getRules() {
         Resource rules = getDataDirectory().get(getRulesPath());
-        return getRules(() -> rules.in());
+        return getRules(rules);
     }
 
     private static GeoServerDataDirectory getDataDirectory() {
         return (GeoServerDataDirectory) GeoServerExtensions.bean("dataDirectory");
     }
 
-    public static List<Rule> getRules(Supplier<InputStream> iss) {
-        try (InputStream inputStream = iss.get()) {
-            if (inputStream.available() == 0) {
-                Utils.info(LOGGER, "Parameters extractor rules file seems to be empty.");
-                return new ArrayList<>();
-            }
+    /**
+     * Read a list of Rule from a resource. Return an empty list if the resource does not exist.
+     * This prevents Resource.in() from creating the file or throwing an exception.
```

Comment:
This comment is no is not part of the api; if you wish to added it below you can.
```suggestion
```